tests/full_tester.c

```diff
@@ -0,0 +1,311 @@
+        printf("Enter the filestem that you want written to>");
+        scanf("%32s", filename);
+        prepend(filename, "./keys/");
+
+        FILE *pub_key = fopen(strcat(filename, ".pub"), "w+");
+        for (unsigned int i = 0; i < XMSS_OID_LEN + params.pk_bytes; i++) {
+            fputc(pk[i], pub_key);
+        }
+        fclose(pub_key);
+
+        // Changing the .pub extension to .prv
+        filename[strlen(filename) - 2] = 'r'; filename[strlen(filename) - 1] = 'v';
+        FILE *prv_key = fopen(filename, "w+");
+        for (unsigned int i = 0; i < sk->length_secret_key; i++) {
+            fputc(sk->secret_key[i], prv_key);
+        }
+        fclose(prv_key);
+
+        
+    } else {
+
+        /* === ACCESSING STORED KEYS === */
+
+        printf("Enter the filestem where the keys are saved> ");
+        scanf("%32s", filename);
+        prepend(filename, "./keys/");   
+        
+        FILE *pub_key = fopen(strcat(filename, ".pub"), "rb");
+        if (pub_key == NULL) return -1;
+        for (unsigned int i = 0; i < params.pk_bytes + XMSS_OID_LEN; i++) {
+            pk[i] = fgetc(pub_key);
+        }
+        fclose(pub_key);
+
+        // Changing the .pub extension to .prv
+        filename[strlen(filename) - 2] = 'r'; filename[strlen(filename) - 1] = 'v';
+
+        printf("Reading in the secret key := \n");
+        FILE *prv_key = fopen(filename, "rb");
+        if (prv_key == NULL) return -1;
+        for (unsigned int i = 0; i < sk->length_secret_key; i++) {
+            sk->secret_key[i] = fgetc(prv_key);
+        }
+        fclose(prv_key);
+        printf("\n");
+
+    }
+
+    #ifdef DEBUGGING
+        // Print out the public key, secret key as part of the debugging process
+        printf("pk="); hexdump(pk, sizeof pk); printf("\n");
+        printf("sk="); hexdump(sk->secret_key, sk->length_secret_key); printf("\n");
+        
+        printf("Continue (0 - no, 1 - yes) ? >");
+        scanf("%d", &decision);
+        if (decision == 0) return -1;   
+    #endif
+
+
+    #ifdef MAX_MOD
+        // Change the max field of the secret key as part of the debugging process
+        unsigned long long number_of_sigs;
+        printf("Enter the max no. of the signatures >");
+        scanf("%llu", &number_of_sigs);
+
+        if (xmss_modify_maximum(sk, number_of_sigs) != 0) {
+            printf("\nError in modifying the maximum number of signatures\n");
+            return -1;
+        }
+        printf("\nnew_sk(post modify)="); hexdump(sk->secret_key, sk->length_secret_key); printf("\n");
+    #endif
+
+    printf("Testing %d %s signatures.. \n", NUM_TESTS, name);
+    for (i = 0; i < NUM_TESTS; i++) {
+        printf("\n\n=========  - iteration #%d: ==============\n", i);
+
+        /* ========================== SIGNING ================================= */
+        randombytes(m, XMSS_MLEN);
+        if(xmssmt){
+            ret = xmssmt_sign(sk, sm, &smlen, m, XMSS_MLEN);
```

Comment:
Can you use `remaining` signature `API` to check if it has run our of signature?